### PR TITLE
Update PR template to encourage covering new functionalities with e2e tests

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Following proposed best practices guide, this PR encourage developers to cover new functionalities with e2e tests.

https://www.bestpractices.dev/en/projects/6593?criteria_level=1#quality
image

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- No tests

- [x] **DONE**

## Links

Port(s): 
- Manager-4.3 https://github.com/SUSE/spacewalk/pull/24591

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ After:
 - [ ] **DONE**
 
 ## Test coverage
+ℹ️ If a major new functionality is added, tests for the new functionality MUST be added to the Cucumber test suite
```
Comment (juliogonzalez) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: IMHO `MUST` is not really encouragement, but enforcement.

Which one do we want? I'd rather go with encouragement for now, so maybe:
```
If a major new functionality is added, it is **strongly recommended** that tests for the new functionality are added to the Cucumber test suite
```
?

Comment (srbarrios) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Ok. If you ask me, I would enforce it, for major new functionality :)
But ok, let's start with a bit less ambition hehe

**Comment by juliogonzalez** [comment]
Remember to backport this later :-)

**Comment by srbarrios** [comment]
@admd feel free to merge this, it only change the template.

**Pull request merged**